usaspending_api/recipient/v2/views/recipients.py

```diff
@@ -225,60 +231,85 @@ def extract_business_categories(recipient_name, recipient_duns, recipient_hash):
+    search.aggs["group_by_recipient"].metric("sum_face_value_loan", sum_face_value_loan)
+    search.aggs["group_by_recipient"].metric("obligation_count", obligation_count)
+    search.aggs["group_by_recipient"].metric("face_value_loan_count", face_value_loan_count)
+
+    response = search.handle_execute()
+    response_as_dict = response.aggs.to_dict()
+    recipient_info_buckets = response_as_dict.get("group_by_recipient", {}).get("buckets", [])
+
+    result_list = []
+
+    for bucket in recipient_info_buckets:
+        result = {}
         if children:
-            # Group by the child recipients
-            queryset = (
-                queryset.values("recipient_hash", "recipient_unique_id", "recipient_name")
-                .annotate(total=Sum("generated_pragmatic_obligation"), count=Sum("counts"))
-                .values("recipient_hash", "recipient_unique_id", "recipient_name", "total", "count")
-            )
-            results = list(queryset)
-        else:
-            # Calculate the overall totals
-            aggregates = queryset.aggregate(total=Sum("generated_pragmatic_obligation"), count=Sum("counts"))
-            aggregates.update({"recipient_hash": recipient_id[:-2]})
-            results = [aggregates]
-    for result in results:
-        result["count"] = result["count"] if result["count"] else 0
-        result["total"] = result["total"] if result["total"] else 0
-    return results
+            recipient_info = json.loads(bucket.get("key"))
+            hash_with_level = recipient_info.get("hash_with_level") or None
+            result = {
+                "recipient_hash": hash_with_level[:-2] if hash_with_level else None,
+                "recipient_unique_id": recipient_info.get("unique_id") or None,
+                "recipient_name": recipient_info.get("name") or None,
+            }
+        result.update(
+            {
+                "total_obligation_amount": int(bucket.get("sum_obligation", {"value": 0})["value"]) / Decimal("100"),
+                "total_obligation_count": bucket.get("obligation_count", {"value": 0})["value"],
+                "total_face_value_loan_amount": int(bucket.get("sum_face_value_loan", {"value": 0})["value"])
+                / Decimal("100"),
+                "total_face_value_loan_count": bucket.get("face_value_loan_count", {"value": 0})["value"],
+            }
+        )
+        result_list.append(result)
+
+    return result_list
 
 
 class RecipientOverView(APIView):
     """
     This endpoint returns a high-level overview of a specific recipient, given its id.
     """
 
-    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/recipient/duns.md"
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/recipient/duns/recipient_id.md"
 
     @cache_response()
     def get(self, request, recipient_id):
-        get_request = request.query_params
-        year = validate_year(get_request.get("year", "latest"))
```

Comment:
Is there a reason `validate_year()` is being removed?

Reply:
Sure isn't, should not be removed so I will put it back